### PR TITLE
Bump angular dependencies

### LINKED ISSUE #9817: Upgrade to Angular 8

### PR DESCRIPTION
**Pull request by vishal423**
Closes #9817 

- [x] Upgrade core-js
- [x] Upgrade ng-jhipster
- [x] Upgrade cache-loader

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/package.json.ejs [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "core-js": "2.6.5",
     "moment": "2.24.0",
     "ng-jhipster": "0.9.3",
-    "ngx-cookie": "4.1.2",
```
Comment (DanielFran) [inline generators/client/templates/angular/package.json.ejs]: Can you try to add @nguniversal/express-engine to peer dependencies?
see: https://github.com/salemdar/ngx-cookie/commit/1956eff8940859147268f5b2b6cc3e7619a89247

Comment (vishal423) [reply generators/client/templates/angular/package.json.ejs]: It was tried before in https://github.com/jhipster/generator-jhipster/pull/9354 and reverted as this peer dependency is more appropriate for server-side rendering. There is an associated open issue on that project that was also linked, but, don't see much progress on that. I am of the opinion to move away from this dependency, but, maybe in separate PR.

**Comment by vishal423** [comment]
`ng-jhipster` updates are tracked under https://github.com/jhipster/ng-jhipster/pull/95

**Comment by wmarques** [comment]
@vishal423 Maybe we can also change the lazy-loading module syntax to the new one using `import()`

File: generators/client/templates/angular/src/test/javascript/jest-global-mocks.ts.ejs [hunk 2]
```diff
@@ -1,15 +1,3 @@
...
-    return {
-        getItem: key => (key in storage ? storage[key] : null),
-        setItem: (key, value) => (storage[key] = value || ''),
-        removeItem: key => delete storage[key],
-        clear: () => (storage = {})
-    };
-};
-
-Object.defineProperty(window, 'localStorage', { value: mock() });
-Object.defineProperty(window, 'sessionStorage', { value: mock() });
```
Comment (wmarques) [inline generators/client/templates/angular/src/test/javascript/jest-global-mocks.ts.ejs]: why removing this? Are they mocked by jest himself?

File: generators/client/templates/angular/package.json.ejs [hunk 3]
```diff
@@ -27,129 +27,129 @@
...
+    "@angular/compiler": "8.0.0",
+    "@angular/core": "8.0.0",
+    "@angular/forms": "8.0.0",
+    "@angular/platform-browser": "8.0.0",
+    "@angular/platform-browser-dynamic": "8.0.0",
+    "@angular/router": "8.0.0",
+    "@fortawesome/angular-fontawesome": "0.4.0",
+    "@fortawesome/fontawesome-svg-core": "1.2.18",
+    "@fortawesome/free-solid-svg-icons": "5.8.2",
+    "@ng-bootstrap/ng-bootstrap": "4.1.3",
```
Comment (vishal423) [inline generators/client/templates/angular/package.json.ejs]: In `v4.2`, date objects are not being considered as valid (validations inbuilt into ng-bootstrap and introduced in `v4.2`). There is an open issue to track this: https://github.com/ng-bootstrap/ng-bootstrap/issues/3215

Reverting to version before that works fine.

File: generators/client/templates/angular/src/test/javascript/jest-global-mocks.ts.ejs (hunk 2 above)
Comment (vishal423) [reply generators/client/templates/angular/src/test/javascript/jest-global-mocks.ts.ejs]: `jsdom` supports web storage APIs, so, no need to mock.

Comment (wmarques) [reply generators/client/templates/angular/src/test/javascript/jest-global-mocks.ts.ejs]: Oh ok, perfect then :)

**Comment by vishal423** [comment]
@wmarques, I need to check on the lazy loading syntax change, specifically for JIT compilation as we use a separate library and I guess that doesn't yet support dynamic imports. I think this change can be made into separate PR.

**Comment by vishal423** [comment]
I spent some time to leverage new dynamic import syntax and observed few issues:
- Due to an issue in `npm` https://npm.community/t/packages-with-peerdependencies-are-incorrectly-hoisted/4794, `AOT` build fails. There is an associated open issue on `webpack` that provide more details: https://github.com/webpack/webpack/issues/8656 . It's also referenced in the angular CLI https://github.com/angular/angular-cli/issues/14566. I tried workaround suggested in the CLI ticket, but, that doesn't seem to work in my sample project.  If you switch to `yarn`, then, AOT build works just fine.
- JIT build compiles with newer syntax, however, I don't see bundle splitting and all lazy loaded modules end up in the main bundle. I didn't debugged further due to AOT build issues.

**Comment by wmarques** [comment]
@vishal423 Ok then, let's merge this and we'll look onto that on a separate PR, thanks for your investigation 👍 
Don't forget to claim your bug bounty ;)

File: .travis.yml [hunk 4]
```diff
@@ -52,6 +52,7 @@ env:
         - JHI_DISABLE_WEBPACK_LOGS=true
         - JHI_E2E_HEADLESS=true
         - JHI_SCRIPTS=$TRAVIS_BUILD_DIR/test-integration/scripts
+        - NG_CLI_ANALYTICS="false"
```
Comment (pascalgrimaud) [inline .travis.yml]: @vishal423 : can you explain me what is it plz ? Maybe we need to add this to Azure Pipelines too ?

**Comment by DanielFran** [comment]
@wmarques Do not forget that it needs a new version of ng-jhipster.
See https://github.com/jhipster/ng-jhipster/pull/95

File: .travis.yml (hunk 4 above)
Comment (wmarques) [reply .travis.yml]: Angular CLI added an analytics feature (like us), this env variable prevents from sending data usage to the CLI project.
I don't think we need to add it into our pipeline as we don't rely on the angular CLI to build our application.

Comment (vishal423) [reply .travis.yml]: In absence of this option, Travis builds got stuck in the post-install scripts (failed build logs @ https://travis-ci.org/jhipster/generator-jhipster/builds/539685496). I haven't noticed this issue on the Azure build pipleines (not sure why such discrepancy).

**Pull request merged**